src/main/java/decodes/cwms/CwmsTimeSeriesDb.java

```diff
@@ -809,19 +809,21 @@ public DbKey connect( String appName, Properties credentials )
 	{
 		String dbUri = this.dbUri != null ? this.dbUri : DecodesSettings.instance().editDatabaseLocation;
 
-		String username = credentials.getProperty("username");
-		String password = credentials.getProperty("password");
-
-		CwmsGuiLogin cgl = CwmsGuiLogin.instance();
-		if (DecodesInterface.isGUI())
-		{
+		String username = null;
+		String password = null;
+		CwmsGuiLogin cgl = null;
+		if (credentials == null && DecodesInterface.isGUI())
+		{			
+			cgl = CwmsGuiLogin.instance();
```

Comment:
comment from somebody new here:

Seeing the word 'Gui' in a database class feels odd.

Reply:
Yeah, that particular code path is rather misplaced. I've recently merged in a more generic support for GUI Login and that will get fixed later. This change was to correct a bug that got introduced why I created the new `AuthSource` ServiceProviderInterface`.